setup.py

```diff
@@ -146,15 +147,15 @@ def build_extension(self, ext):
                 os.environ['_PYTHON_HOST_PLATFORM'] = re.sub(r'macosx-[0-9]+\.[0-9]+-(.+)', r'macosx-10.9-\1', util.get_platform())
 
             # Specify how many threads to use when building, depending on available memory
-            if totalMemory < 1000:
-                build_args += ['--', '-j1']
-                cmake_args += ['-DHUNTER_JOBS_NUMBER=1']
-            elif totalMemory < 2000:
-                build_args += ['--', '-j2']
-                cmake_args += ['-DHUNTER_JOBS_NUMBER=2']
-            else:
-                # Build with maximum available threads
-                build_args += ['--', '-j']
+            max_threads = multiprocessing.cpu_count()
+            num_threads = (freeMemory // 1000)
+            num_threads = min(num_threads, max_threads)
+            if num_threads <= 0:
+                num_threads = 1            
+            build_args += ['--', '-j' + str(num_threads)]
+            if num_threads <= 2:
+                cmake_args += ['-DHUNTER_JOBS_NUMBER=' + str(num_threads)]
```

Comment:
Hunter should be set with same number of threads as CMake build
```
build_args += ['--', '-j' + str(num_threads)]
cmake_args += ['-DHUNTER_JOBS_NUMBER=' + str(num_threads)]
```

Reply:
Ohh okay. The previous code had only two max so kept that part the same. will change it.

Reply:
Doesn't that also make twice the threads. like 2 threads per process..

Reply:
No - Hunter executes first, then build happens. So no overlap occurs.